### PR TITLE
Include C++ solution for usaco-831

### PR DESCRIPTION
**Pull request by Aadit-Ambadkar**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-831.mdx [hunk 1]
```diff
@@ -3,14 +3,117 @@ id: usaco-831
 source: USACO Bronze 2018 Open
 title: Team Tic Tac Toe
 author: Jesse Choe, Kevin Sheng, Benjamin Qi, Brad Ma
+contributor: Aadit Ambadkar
```
Comment (ryanchou-dev) [inline solutions/usaco-831.mdx:6]: contributors don't work on sols, add yourself as an author

File: solutions/usaco-831.mdx [hunk 2]
```diff
@@ -3,14 +3,117 @@ id: usaco-831
...
+
+	return 0;
+}
+
+int main() {
+	freopen("tttt.in", "r", stdin);
+	freopen("tttt.out", "w", stdout);
 	
+	for (int i = 0; i < 3; i++) {
+		string line; cin >> line;
```
Comment (ryanchou-dev) [inline solutions/usaco-831.mdx:94]:
```suggestion
		string line;
		cin >> line;
```

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

File: solutions/usaco-831.mdx [hunk 3]
```diff
@@ -3,14 +3,116 @@ id: usaco-831
...
+		return false;
+	}
+	if (a != ch2 && b != ch2 && c != ch2) {
+		return false;
+	}
+
+	return true;
+}
+// Does a team win?
+int teamWins(char ch1, char ch2) {
```
Comment (SansPapyrus683) [inline solutions/usaco-831.mdx:68]: c++ uses snake_case, best to be consistent

File: solutions/usaco-831.mdx [hunk 4]
```diff
@@ -168,6 +279,8 @@ int main() {
 	written << winners[2].size() << endl;
 }
 ```
-
 </CPPSection>
+<JavaSection>
+Unfortunately, implementing this solution in java is quite tricky, and is beyond the scope of Bronze
```
Comment (SansPapyrus683) [inline solutions/usaco-831.mdx:284]: bruh

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-831.mdx (hunk 4 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-831.mdx:284]: what? You need custom comparators for Java with sorted set...

File: solutions/usaco-831.mdx (hunk 3 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-831.mdx:68]: ok

File: solutions/usaco-831.mdx (hunk 1 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-831.mdx:6]: ok

File: solutions/usaco-831.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-831.mdx:284]: i mean what's wrong with that

Comment (Aadit-Ambadkar) [reply solutions/usaco-831.mdx:284]: It's not in the scope of bronze

Comment (Aadit-Ambadkar) [reply solutions/usaco-831.mdx:284]: image

Comment (SansPapyrus683) [reply solutions/usaco-831.mdx:284]: so?

Comment (Aadit-Ambadkar) [reply solutions/usaco-831.mdx:284]: ik but it's better to not include that in this solution.

Comment (SansPapyrus683) [reply solutions/usaco-831.mdx:284]: > this

this solution? what other solution file could there be?

File: solutions/usaco-831.mdx (hunk 3 above)
Comment (bqi343) [reply solutions/usaco-831.mdx]: fwiw [google's style guide](https://google.github.io/styleguide/cppguide.html#Function_Names) begs to differ

File: solutions/usaco-831.mdx [hunk 5]
```diff
@@ -2,15 +2,117 @@
...
 
 ## Solution 1
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_tttt_bronze_open18.html)
 
 <LanguageSection>
+<CPPSection>
 	
+```cpp
+#include <bits/stdc++.h>
```
Comment (bqi343) [inline solutions/usaco-831.mdx:16]: I don't see a need to include this; it's not more readable than the original code in the analysis.

File: solutions/usaco-831.mdx [hunk 6]
```diff
@@ -145,20 +253,23 @@ void insert(vector<pair<int, int>> coordinates) {
 	winners[contained.size()].insert(contained);
 }
 
-// 2018 usopen bronze
 int main() {
 	ifstream read("tttt.in");
+
```
Comment (bqi343) [inline solutions/usaco-831.mdx:258]: no need to add these newlines

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/usaco-831.mdx (hunk 4 above)
Comment (ryanchou-dev) [inline solutions/usaco-831.mdx:284]: bruh.

tbh, it's better to add the solution and an info block saying that it uses concepts from silver.

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
remove unnecessary new lines (see benq's comments)

rest is ok

File: solutions/usaco-831.mdx (hunk 4 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-831.mdx]: This probably isn't the right pr to even include a java solution tbh. I can create another one for that.

File: solutions/usaco-831.mdx (hunk 5 above)
Comment (bqi343) [reply solutions/usaco-831.mdx:16]: ^

File: solutions/usaco-831.mdx [hunk 7]
```diff
@@ -145,20 +253,22 @@ void insert(vector<pair<int, int>> coordinates) {
...
 	for (int r = 0; r < WIDTH; r++) {
 		read >> board[r];
 	}
+	
 	// insert rows
 	for (int i = 0; i < WIDTH; i++) {
 		insert({{i, 0}, {i, 1}, {i, 2}});
 	}
-	// insert columns
+
```
Comment (bqi343) [inline solutions/usaco-831.mdx:266]: again, none of these additional newlines are necessary

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**